### PR TITLE
Webpack

### PR DESCRIPTION
**Pull request by m0g**
Several things for that PR:

* `superdesk-client-core` has been updated v1.5.1
* Travis is now running against ubuntu trusty.
* The style has been migrated from `less` to `sass`
* We are now using Webpack for building the frontend.
* The code base is slowly moving to ES2015 with ES2015 modules (import/export syntax).
* Updating protactor from v2 to v4

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -51,7 +51,7 @@ COPY ./client/package.json /opt/superdesk/client/
 RUN cd ./client && npm install
 COPY ./client/bower.json /opt/superdesk/client/
 COPY ./client/.bowerrc /opt/superdesk/client/
-RUN cd ./client && bower --allow-root install
+#RUN cd ./client && bower --allow-root install
```
Comment (nistormihai) [inline Dockerfile]: remove comment.

File: client/Gruntfile.js [hunk 2]
```diff
@@ -40,50 +39,46 @@ module.exports = function (grunt) {
...
-        'watch'
+        'copy:assets',
+        'copy:index',
+        'copy:sirTrevor',
+        'ngtemplates:dev',
+        'webpack-dev-server:start'
     ]);
+
+    grunt.registerTask('ci:travis', []);
+    //grunt.registerTask('ci:travis', [
```
Comment (nistormihai) [inline client/Gruntfile.js]: remove comment

File: client/app/index.html [hunk 3]
```diff
@@ -80,191 +70,7 @@ <h3 translate>About</h3>
...
-            'liveblog-edit/directives',
-            'superdesk-settings/module',
-            'superdesk-dashboard/module',
-            'superdesk-archive/module',
-            'superdesk-ingest/module'
-            ], function(bootstrap) {
-                bootstrap(config, modules);
-            });
-    </script>
+    <script src="sir-trevor.js"></script>
```
Comment (nistormihai) [inline client/app/index.html]: waw finally clean :100:

File: client/app/scripts/index.js [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+import 'superdesk-core/scripts/apps/highlights';
+import 'superdesk-core/scripts/apps/translations';
+import 'superdesk-core/scripts/apps/content-filters';
+import 'superdesk-core/scripts/apps/dictionaries';
+import 'superdesk-core/scripts/apps/vocabularies';
+import 'superdesk-core/scripts/apps/archive';
+import 'superdesk-core/scripts/apps/monitoring';
+import 'superdesk-core/scripts/apps/settings';
+import 'superdesk-core/scripts/apps/ingest';
+//import 'superdesk-core/scripts/apps/search-providers';
```
Comment (nistormihai) [inline client/app/scripts/index.js]: remove comment

File: client/app/scripts/liveblog-edit/module.js [hunk 5]
```diff
@@ -608,9 +634,9 @@ define([
         'angular-embed',
         'angular-embed.handlers',
         'ngRoute',
-        'superdesk.services.modal',
-        'superdesk.upload',
-        'superdesk.editor',
+        'superdesk.core.services.modal',
+        'superdesk.core.upload',
+        //'superdesk.core.editor3',
```
Comment (nistormihai) [inline client/app/scripts/liveblog-edit/module.js]: remove comment.

File: client/app/scripts/liveblog-edit/posts.service.js [hunk 6]
```diff
@@ -7,12 +7,15 @@
...
+import _ from 'lodash';
+import moment from 'moment';
 
-define([
-    'angular',
-    'moment',
-    'lodash'
-], function(angular, moment, _) {
+//define([
+//    'angular',
```
Comment (nistormihai) [inline client/app/scripts/liveblog-edit/posts.service.js]: remove comment

File: client/Gruntfile.js (hunk 2 above)
Comment (m0g) [inline client/Gruntfile.js]: About this one, all the linting should go in there. But I think in the near future, we'll just use the superdesk eslint configuration

File: client/app/scripts/index.js (hunk 4 above)
Comment (m0g) [inline client/app/scripts/index.js]: This is actually quite helpful, it allows us to keep track of what superdesk modules we can safely disable without breaking the app

File: client/app/scripts/liveblog-edit/module.js (hunk 5 above)
Comment (m0g) [inline client/app/scripts/liveblog-edit/module.js]: same here

File: client/app/index.html (hunk 3 above)
Comment (m0g) [inline client/app/index.html]: :+1:

File: client/package.json [hunk 7]
```diff
@@ -12,8 +12,14 @@
     }
   ],
   "devDependencies": {
+    "babel-core": "^6.10.4",
```
Comment (nistormihai) [inline client/package.json]: this is not needed in `devDependencies` it is loaded from [superdesk](https://github.com/superdesk/superdesk-client-core/blob/v1.5.1/package.json#L45)

File: client/app/scripts/index.js (hunk 4 above)
Comment (nistormihai) [inline client/app/scripts/index.js]: please specify this with an comment to be more clear why they are there.

File: README.md [hunk 8]
```diff
@@ -39,7 +39,7 @@ echo "discovery.zen.ping.multicast.enabled: false" | sudo tee --append /etc/defa
 Install Node.js LTS version:
 
 ```bash
-curl -sL https://deb.nodesource.com/setup_4.x | sudo -E bash -
+curl -sL https://deb.nodesource.com/setup_6.x | sudo -E bash -
```
Comment (nistormihai) [inline README.md]: bower installation should be removed

File: client/app/scripts/index.js (hunk 4 above)
Comment (m0g) [inline client/app/scripts/index.js]: You a comment to comment the comment?

File: client/package.json (hunk 7 above)
Comment (m0g) [inline client/package.json]: Probably it should be in `dependencies`, but it is not loaded from superdesk

File: README.md (hunk 8 above)
Comment (m0g) [inline README.md]: https://dev.sourcefabric.org/browse/LBSD-1263

**Pull request merged**